### DISCUSSION TITLE
SurrealDB as storage

### DISCUSSION THREAD

**Discussion by dvv** [kanidm/kanidm#2642] [Ideas]
Hi!

I wonder if we could explore the subj?

They have [the same thing](https://surrealdb.com/docs/surrealdb/introduction/architecture#tikv) for storage.
The pros are to have human-readable auto-migrating storage definition with [controllable access to fields](https://surrealdb.com/docs/surrealdb/surrealql/statements/define/field#setting-permissions-on-fields), built-in [event system](https://surrealdb.com/docs/surrealdb/surrealql/statements/define/event#example-usage) etc.

TIA

**Comment by Firstyear** [comment]
We will never support any other backends beside sqlite or another local kv store. We *must* control replication because we have specific and specialised requirements for attribute handling in conflict cases.
